index.js

```diff
@@ -1,10 +1,22 @@
 'use strict';
+var Promise = require('pinkie-promise');
 var arrayUnion = require('array-union');
 var objectAssign = require('object-assign');
-var async = require('async');
 var glob = require('glob');
 var arrify = require('arrify');
 
+function globP(pattern, opts) {
+	return new Promise(function (resolve, reject) {
+		glob(pattern, opts, function (err, files) {
+			if (err) {
+				reject(err);
```

Comment:
Might be good to `return` after rejecting.
Although ES2015 promises' can't mutate their state after being settled (rejected or fulfilled), it looks odd to call both `reject` and `resolve` when there's an error.